### ISSUE TITLE
How to use this

### ISSUE THREAD

**Issue by waheedshahani** [tplgy/tgl#2]
Is there any sample code to use this tgl? I am more interested in getting updates on my channel, is this implemented already?

**Comment by ChristianMertes** [comment]
Yeah, I want to second this! Just a couple lines of code: how to set up a client, auth a user and send two or three API calls would already be much appreciated as a starting point!

**Comment by ChristianMertes** [comment]
Although, after some more searching, there is the telegram-cli which uses the tgl library this is a fork of. So basically, the starting point is https://github.com/vysheng/tg/blob/master/main.c

**Comment by ChristianMertes** [comment]
Sorry, having a bit of a monologue here. Seems like those two different projects both named tgl have absolutely nothing in common. So no, there's still no starting point. Also, this is really confusing!

**Comment by jpetso** [comment]
[ex-coworker here]
This project is indeed a fork of the abandoned https://github.com/vysheng/tgl which explains the common name. However, due to code quality concerns of the original codebase, @Bomb3rman (with @leoyanggit and @efidler)  decided to refactor the code extensively and give it a new C++ API, while also fixing bugs and extending its functionality.

So the overall workings should still be similar, and better, although I haven't used it myself so I can't help with actual examples (sorry).

**Comment by ChristianMertes** [comment]
Thanks for the reply! That's helpful to know :)

As for how to use it, while I haven't done a minimal example yet, the starting point seems to be tgl_user_agent::create() in include/tgl/tgl_user_agent.h seems to be a good starting point to get a handle of the code. Once I produce some actually working code, I'm going to share it here.

**Comment by kenkit** [comment]
We all know that C/C++ is the most powerful language, we could use a sample and some documentation to get things going.
The easier it is to use, the more it will improve and gain the necessary support.

**Comment by sergiokovalski** [comment]
Can I use this lib with telegram-cli?
Replace default vysheng tgl?
